### ISSUE TITLE
FC0002 - Wrong Warning for purchaseHeader

### ISSUE THREAD

**Issue by Arthurvdv** [ALCops/Analyzers#307]
I now receive several warnings like the following:

> Wrong casing detected! Use purchaseHeader instead.

<img width="610" height="274" alt="Image" src="https://github.com/user-attachments/assets/81be7b1f-a3b3-4a11-adfb-da34f3175f61" />

Was anything changed here?

---
### Comment by @Arthurvdv on 2025-09-09T14:21:05Z

No, not deliberately, at least 🤔

Do you have a small code sample for me?

### Comment by @pri-kise on 2025-10-06T07:26:45Z

I could reproduce the issue.
```al
    procedure Dummy(var PurchaseHeader: Record "Purchase Header")
    begin
        MyProcedure1(PurchaseHeader);
        MyProcedure2(PurchaseHeader);
        MyProcedure3(PurchaseHeader);
    end;

    local procedure MyProcedure1(PurchaseHeader: Record "Purchase Header")
    begin
        if PurchaseHeader."No." = '' then
            exit;
    end;

    local procedure MyProcedure2(PurchaseHeader: Record "Purchase Header")
    begin
        if PurchaseHeader."No." = '' then
            exit;
    end;

    local procedure MyProcedure3(purchaseHeader: Record "Purchase Header")
    begin
        if PurchaseHeader."No." = '' then
            exit;
    end;
```

The wrong warning only occurs when the last procedure of a codeunit has a wrong casing.

I don't know why but then the warning is shown for any previous procedure.

### Comment by @Arthurvdv on 2025-10-07T08:14:07Z

Thanks for the code example, I think I have an idea on why this is going wrong. I'll look into this!

---
*Copied from original issue: https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/1168*

**Comment by Arthurvdv** [comment]
Hi @pri-kise,

This one took a bit longer than expected, but I finally(!) managed to find a solution and I believe the issue has now been resolved in ALCops version `0.9.0-beta.2`.

Feel free to try out the beta version and let me know how it works for you.

**Comment by Arthurvdv** [comment]
Hi @pri-kise,

The fix for this issue was previously released in the beta channel and is now available in the latest stable release of ALCops, version `0.9.0`.

I'm going to close this issue for now. If you're still experiencing problems or notice anything that doesn't behave as expected, please feel free to reopen it.
